onadata/apps/api/permissions.py

```diff
@@ -560,11 +560,26 @@ def has_permission(self, request, view):
         return True
 
 
-class EntityListPermission(DjangoObjectPermissionsAllowAnon):
+class EntityListPermission(DjangoObjectPermissions):
     """Permission for EntityList"""
 
+    authenticated_users_only = False
+
     def has_permission(self, request, view):
-        if request.user.is_anonymous:
+        if request.user.is_anonymous and view.action not in ["create"]:
             return True
 
+        if request.user.is_authenticated and view.action == "create":
+            project_id = request.data.get("project")
+
+            if project_id:
+                try:
+                    project = Project.objects.get(pk=int(project_id))
+
+                except (ValueError, Project.DoesNotExist):
+                    # Return as validation will take care of this
```

Comment:
Does returning false cause issues a this stage - could we handle the false that is returned?

Reply:
@ukanga If we return `False`, this will return status `403`. A person who is authorized to create and makes a request with an invalid `project`, they should get a `400` response instead of a`403`. I chose to return `True` so that request will reach the serializer and a `400` error is returned.

Maybe for code clarity, we should raise a validation error at this stage?

Reply:
@ukanga I have refactored to raise validation error message

Reply:
Why is the permission check happening before the validation check? Is there a different workflow we should consider?

Reply:
@ukanga The idea is to return 400 (Project does not exist) if the project literally does not exist, or if the user has no permission to view the project. If the user has permission to view the project but does not have permission to create a dataset for the project, they get 403. 

Maybe I need to add better comments to communicate the message above